src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1,855 +1,245 @@
-    </footer>
-  </body>
-</html>
-```
-
-### `Astro.slots`
+The `Astro` global object is available to all `.astro` files. Use the `context` object in [endpoint functions](/en/guides/endpoints/) to serve static or live server endpoints and in [middleware](/en/guides/middleware/) to inject behavior when a page or endpoint is about to be rendered.
 
-`Astro.slots` contains utility functions for modifying an Astro component's slotted children.
+## The context object
 
-#### `Astro.slots.has()`
+The following items are available on the `Astro` global (e.g. `Astro.props`, `Astro.redirect()`) and most are also available on an endpoint context object (e.g. `context.props`, `context.redirect()`).
 
-<p>
-
-**Type:** `(slotName: string) => boolean`
-</p>
+### `props`
 
-You can check whether content for a specific slot name exists with `Astro.slots.has()`. This can be useful when you want to wrap slot contents, but only want to render the wrapper elements when the slot is being used.
+`props` is an object containing any values that have been passed as [component attributes](/en/basics/astro-components/#component-props).
 
-```astro  title="src/pages/index.astro"
+```astro {3}
 ---
+// src/components/Heading.astro
+const { title, date } = Astro.props;
 ---
-<slot />
-
-{Astro.slots.has('more') && (
-  <aside>
-    <h2>More</h2>
-    <slot name="more" />
-  </aside>
-)}
+<div>
+  <h1>{title}</h1>
+  <p>{date}</p>
+</div>
 ```
 
-#### `Astro.slots.render()`
-
-<p>
-
-**Type:** `(slotName: string, args?: any[]) => Promise<string>`
-</p>
-
-You can asynchronously render the contents of a slot to a string of HTML using `Astro.slots.render()`.
-
-```astro
+{/* ```astro /title=".+"/ /date=".+"/ */}
+```astro "title=" "date="
 ---
-const html = await Astro.slots.render('default');
+// src/pages/index.astro
+import Heading from '../components/Heading.astro';
 ---
-<Fragment set:html={html} />
+<Heading title="My First Post" date="09 Aug 2022" />
 ```
 
-:::note
-This is for advanced use cases! In most circumstances, it is simpler to render slot contents with [the `<slot />` element](/en/basics/astro-components/#slots).
-:::
+<ReadMore>Learn more about how [Markdown and MDX layouts](/en/guides/markdown-content/#frontmatter-layout-property) handle props.</ReadMore>
+
+
+The `props` object also contains any `props` passed from `getStaticPaths()` when rendering static routes.
+
+<Tabs>
+  <TabItem label="Astro.props">
+    ```astro title="src/pages/posts/[id].astro" "props:" {11}
+    ---
+    export function getStaticPaths() {
+      return [
+        { params: { id: '1' }, props: { author: 'Blu' } },
+        { params: { id: '2' }, props: { author: 'Erika' } },
+        { params: { id: '3' }, props: { author: 'Matthew' } }
+      ];
+    }
+
+    const { id } = Astro.params;
+    const { author } = Astro.props;
+    ---
+    ```
+  </TabItem>
+  <TabItem label="context.props">
+    ```ts title="src/pages/posts/[id].json.ts" "props:" {11-14}
```

Comment:
nit: currently for the `context.props` example, everything is highlighted for `export function GET` except the closing brace. I think this should be included to avoid confusion if someone tries to copy only the highlighted part:
```suggestion
    ```ts title="src/pages/posts/[id].json.ts" "props:" {11-15}
```